File: extract.js
const fs = require('fs');
const readline = require('readline');

async function processLineByLine() {
  const fileStream = fs.createReadStream('C:\\\\Users\\\\USER\\\\.gemini\\\\antigravity-ide\\\\brain\\\\8a78ab36-08dc-4d7f-b9f9-24df2c110d24\\\\.system_generated\\\\logs\\\\transcript_full.jsonl');

  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity
  });

  let lastCodeContent = null;

  for await (const line of rl) {
    try {
      const data = JSON.parse(line);
      if (data.tool_calls) {
        for (const call of data.tool_calls) {
          if (call.args && call.args.TargetFile && call.args.TargetFile.includes('DashboardSection.tsx')) {
            if (call.args.CodeContent) {
              lastCodeContent = call.args.CodeContent;
            }
          }
        }
      }
    } catch (e) {
      // ignore parse errors
    }
  }

  if (lastCodeContent) {
    fs.writeFileSync('d:\\Devpulse\\extracted_dashboard.tsx', lastCodeContent, 'utf-8');
    console.log('Extracted file successfully!');
  } else {
    console.log('No write_to_file found for DashboardSection.tsx');
  }
}

processLineByLine();

File: extract.py
import json
import re

with open(r'C:\Users\USER\.gemini\antigravity-ide\brain\8a78ab36-08dc-4d7f-b9f9-24df2c110d24\dashboard_calls.jsonl', 'r', encoding='utf-8', errors='ignore') as f:
    lines = f.readlines()

for line in lines:
    match = re.search(r'({.*})', line)
    if match:
        try:
            data = json.loads(match.group(1))
            if 'tool_calls' in data:
                for call in data['tool_calls']:
                    if 'args' in call:
                        args = call['args']
                        if 'TargetFile' in args and 'DashboardSection.tsx' in args['TargetFile']:
                            if 'CodeContent' in args:
                                print(f"Found write_to_file! Length: {len(args['CodeContent'])}")
                                with open('extracted_dashboard.tsx', 'w', encoding='utf-8') as out:
                                    out.write(args['CodeContent'])
                            elif 'ReplacementChunks' in args:
                                print("Found ReplacementChunks")
                            elif 'ReplacementContent' in args:
                                print("Found ReplacementContent")
        except Exception as e:
            pass
